snap/container.go

```diff
@@ -33,7 +33,10 @@ import (
 // Container is the interface to interact with the low-level snap files
 type Container interface {
 	// ReadFile returns the content of a single file from the snap.
-	ReadFile(relative string) (content []byte, err error)
+	ReadFile(relative string) ([]byte, error)
+
+	// ReadDir returns the content of a single directory inside the snap.
+	ReadDir(path string) ([]os.FileInfo, error)
```

Comment:
Can we avoid os.FileInfo here?  With that we force a pretty particular implementation of this method.